peer.go

```diff
@@ -86,8 +86,10 @@ func (p *Peer) Stop() {
 		p.logger.Debug("Stopping peer")
 		close(p.stop) // stops sendLoop and readLoop and statLoop
 		// sendLoop closes p.conn and p.send in defer
-
-		p.net.controller.peerStatus <- peerStatus{peer: p, online: false}
+		select {
+		case p.net.controller.peerStatus <- peerStatus{peer: p, online: false}:
+		case <-p.net.stopper:
+		}
```

Comment:
Shouldn't this rather be:
```golang
select {
  case <-p.net.stopper:
  default:
      p.net.controller.peerStatus <- peerStatus{peer: p, online: false}:
}
```
if `p.net.stopper` is not closed then proceed to the peer un-registration?

Reply:
No, if it's written that way, the line inside default will block until a signal is received without reacting to the stopper channel

Reply:
Sure, in case `peerStatus` would be full that could be a problem...